frontend/public/components/catalog/catalog-items.tsx

```diff
@@ -180,44 +258,42 @@ export const groupItems = (items, groupBy) => {
   return items;
 };
 
-export class CatalogTileViewPage extends React.Component {
+export class CatalogTileViewPage extends React.Component<
+  CatalogTileViewPageProps,
+  CatalogTileViewPageState
+> {
   constructor(props) {
     super(props);
 
     this.state = { detailsItem: null };
-
-    this.openOverlay = this.openOverlay.bind(this);
-    this.closeOverlay = this.closeOverlay.bind(this);
-    this.renderTile = this.renderTile.bind(this);
   }
 
   componentDidMount() {
     const { items } = this.props;
-    const searchParams = new URLSearchParams(window.location.search);
```

Comment:
nit: not necessary to give a type on an assignment like this. It will be inferred.

Reply:
removed in the line